# Rules file for STAR alignment from extracted fastq files.
#
# @zgr2788


# Rule for single-end reads
rule alignSingle:
	input:
		"Single/{runID}/{runID}.fastq"

	output:
		"Single/{runID}/{runID}.bam"

	params:
		starDir = STARidx

	threads: 8

	resources:
		mem_mb = 49152

	shell:
		"""
	 	 module load star-2.7.0e-gcc-9.2.0-vynasg3

	 	 STAR \
	 	 --genomeDir {params.starDir} \
	 	 --readFilesIn {input} \
	 	 --outFileNamePrefix Single/{wildcards.runID}/ \
	 	 --runThreadN {threads} \
	 	 --outBAMsortingThreadN {threads} \
	 	 --outSAMtype BAM SortedByCoordinate \
	 	 --genomeLoad LoadAndRemove \
	 	 --limitBAMsortRAM 50000000000

	 	 mv Single/{wildcards.runID}/Aligned.sortedByCoord.out.bam Single/{wildcards.runID}/{wildcards.runID}.bam
		"""


# Rule for paired-end reads
rule alignPaired:
	input:
		reads_1 = "Paired/{runID}/{runID}_1.fastq",
		reads_2 = "Paired/{runID}/{runID}_2.fastq"

	output:
		"Paired/{runID}/{runID}.bam"

	params:
		starDir = STARidx

	threads: 8

	resources:
		mem_mb = 49152

	shell:
		"""
	 	 module load star-2.7.0e-gcc-9.2.0-vynasg3

	 	 STAR \
	 	 --genomeDir {params.starDir} \
	 	 --readFilesIn  {input.reads_1} {input.reads_2} \
	 	 --outFileNamePrefix  Paired/{wildcards.runID}/ \
	 	 --runThreadN {threads} \
	 	 --outBAMsortingThreadN {threads} \
	 	 --outSAMtype BAM SortedByCoordinate \
	 	 --genomeLoad LoadAndRemove \
	 	 --limitBAMsortRAM 50000000000

	 	 mv Paired/{wildcards.runID}/Aligned.sortedByCoord.out.bam Paired/{wildcards.runID}/{wildcards.runID}.bam
		"""		
